### PR TITLE
LUSD PSM

### PR DESCRIPTION
**Pull request by ElliotFriedman**

### CODE REVIEW

File: contract-addresses/mainnetAddresses.ts [hunk 1]
```diff
@@ -506,6 +506,11 @@ const MainnetAddresses: MainnetAddresses = {
     address: '0xe61d11ec732d556A26fb863B192052BEa03eF8B5',
     category: AddressCategory.Oracle
   },
+  chainlinkLUSDOracle: {
```
Comment (Joeysantoro) [inline contract-addresses/mainnetAddresses.ts:509]: please put external addresses in the external category

File: test/integration/tests/lusdPSM.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+const oneEth = ethers.constants.WeiPerEther;
+const toBN = ethers.BigNumber.from;
+
+before(async () => {
+  chai.use(CBN(ethers.BigNumber));
+  chai.use(solidity);
+  await resetFork();
+});
+
+describe('lusd PSM', function () {
```
Comment (Joeysantoro) [inline test/integration/tests/lusdPSM.ts:38]: I don't think this should be an entirely new integration test, can we move it to the peg one or something?

**Review by Joeysantoro — COMMENTED** [review]
Looks about right, I'll do a deeper review in the morning after you deploy and add the FeiSkimmer (should we add skim logic to the PSM?)

File: proposals/dao/fip_63.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  }
+
+  // 1. Deploy eth PSM
+  const lusdPSM = await (
+    await ethers.getContractFactory('MintRedeemPausePSM')
+  ).deploy(
+    {
+      coreAddress: core,
+      oracleAddress: chainlinkLUSDOracleWrapper,
+      backupOracle: chainlinkLUSDOracleWrapper,
```
Comment (Joeysantoro) [inline proposals/dao/fip_63.ts:51]: leave backup address 0

File: contract-addresses/mainnetAddresses.ts (hunk 1 above)
Comment (ElliotFriedman) [reply contract-addresses/mainnetAddresses.ts:509]: this didn't need to get added. removing

File: test/integration/tests/lusdPSM.ts (hunk 2 above)
Comment (ElliotFriedman) [reply test/integration/tests/lusdPSM.ts:29]: every other psm had it's own IT, this shouldn't be any different

File: proposals/dao/fip_63.ts (hunk 3 above)
Comment (ElliotFriedman) [reply proposals/dao/fip_63.ts:51]: in previous PSM's, the backup address is the same as the main one.

Comment (Joeysantoro) [reply proposals/dao/fip_63.ts:51]:
```
        if (_backupOracle != address(0) && _backupOracle != _oracle) {
            _setBackupOracle(_backupOracle);
        }
```
won't do anything. Fine either way its < 100 gas to do the comparison

File: test/integration/tests/lusdPSM.ts (hunk 2 above)
Comment (Joeysantoro) [reply test/integration/tests/lusdPSM.ts:29]: I think its a bad pattern and am making this suggestion to change, before entrenching a suboptimal solution. Lets discuss from first principles not anchor to precedent. What do you think?

File: proposals/description/fip_63.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  ],
+  description: `
+  This proposal operationalizes the LUSD PSM:
+  1. Add the LUSD PSM to the CR Oracle
+  2. Pause LUSD PCVDripController
+  3. Pause minting and redemptions on the newly created lusd PSM
+  4. Grant the LUSD PSM the minter role
+  5. Grant PCV Controller to the lusdPCVDripController and lusdPSMFeiSkimmer, then pause
+
+  Code:
```
Comment (Joeysantoro) [inline proposals/description/fip_63.ts:84]: Link code address

File: proposals/dao/fip_63.ts (hunk 3 above)
Comment (ElliotFriedman) [reply proposals/dao/fip_63.ts]: gas optimization is a good reason to not lindy this one

File: proposals/description/fip_63.ts (hunk 4 above)
Comment (ElliotFriedman) [reply proposals/description/fip_63.ts]: done

File: test/integration/tests/lusdPSM.ts (hunk 2 above)
Comment (ElliotFriedman) [reply test/integration/tests/lusdPSM.ts:29]: what file are you proposing we move this into? psm.ts?

**Review by Joeysantoro — APPROVED** [review]

**Pull request merged**